src/design-system/components/checkbox/CheckBox.tsx

```diff
@@ -24,10 +31,18 @@ const CheckBox: FC<CheckBoxProps> = ({
   size = 'md',
   ...props
 }) => {
-  const [checked, setChecked] = useState<boolean>(false)
+  const [internalChecked, setInternalChecked] = useState<boolean>(false)
```

Comment:
Add `defaultIsChecked` prop and use it here. And also rename state to `internalIsChecked`, please
```suggestion
  const [internalChecked, setInternalChecked] = useState<boolean>(defaultIsChecked ?? false)
```